docs/aws/audit/elbmonitoring/rules/clb_desync_mode_check.mdx

```diff
@@ -23,6 +23,211 @@ CBP
+
+   ```
+   pip install awscli
+   ```
+   Then configure it using the following command and provide your AWS credentials (Access Key ID and Secret Access Key), default region name, and default output format when prompted.
+
+   ```
+   aws configure
+   ```
+
+2. List all the Load Balancers: Use the following command to list all the Classic Load Balancers (CLBs) in your account.
+
+   ```
+   aws elbv2 describe-load-balancers
+   ```
+
+3. Describe Load Balancer Attributes: For each Load Balancer, describe its attributes using the following command. Replace 'your-load-balancer-arn' with the ARN of your Load Balancer.
+
+   ```
+   aws elbv2 describe-load-balancer-attributes --load-balancer-arn your-load-balancer-arn
+   ```
+
+4. Check Desync Mitigation Mode: In the output of the above command, look for the 'routing.http.desync_mitigation_mode' attribute. If its value is not 'defensive' or 'strictest', then Desync Mitigation Mode is not enabled.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3):
+   You need to install and configure AWS SDK for Python (Boto3) to interact with AWS services. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials either by setting environment variables `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` and `AWS_SESSION_TOKEN`, or by creating the credentials file at `~/.aws/credentials`.
+
+2. Import necessary libraries and create a session:
+   Import the necessary libraries and create a session using your AWS credentials. Here is a sample script:
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       aws_session_token='SESSION_TOKEN',
+   )
+   ```
+
+3. Create an ELB client and retrieve all load balancers:
+   Use the session to create an ELB client. Then, retrieve all load balancers and iterate over them to check their Desync Mitigation Mode. Here is a sample script:
+   ```python
+   elb = session.client('elbv2')
+
+   load_balancers = elb.describe_load_balancers()
+   for lb in load_balancers['LoadBalancers']:
+       attributes = elb.describe_load_balancer_attributes(LoadBalancerArn=lb['LoadBalancerArn'])
+       for attr in attributes['Attributes']:
+           if attr['Key'] == 'routing.http.desync_mitigation_mode':
+               print(f"Load Balancer: {lb['LoadBalancerName']}, Desync Mitigation Mode: {attr['Value']}")
+   ```
+   This script will print the name of each load balancer and its Desync Mitigation Mode.
+
+4. Analyze the output:
+   If the Desync Mitigation Mode of a load balancer is not 'defensive' or 'strictest', it means that the load balancer is misconfigured. You should see an output similar to this for each load balancer:
+   ```
+   Load Balancer: my-load-balancer, Desync Mitigation Mode: defensive
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>